languages/c/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,59 @@
-# How to implement an C concept exercise
+# How to Implement a C Concept Exercise
 
-TODO: describe how to implement a concept exercise for the C track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a concept exercise for the C track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `function-pointers`).
+- `<NAME>`: the name of the exercise in snake_case (e.g. `function_pointers`).
+
+Before implementing the exercise, please make sure you have a good understanding  of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+To implement a concept exercise, the following files must be created:
+<pre>
+languages
+└── c
+    └── exercises
+        └── concept
+            └── &lt;SLUG&gt;
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   ├── hints.md
+                │   └── after.md (optional)
+                ├── .meta
+                │   ├── config.json
+                │   ├── design.md
+                │   ├── example.c
+                │   └── example.c
+                ├── src
```

Comment:
Not sure if this directory is needed. It exists in the current v2 structure, but perhaps the <NAME>{.c | .h} files can now reside directly in the <SLUG> directory?

Reply:
This is something that is for the C maintainers to decide. It is possible to change the structure, which probably influences how the test runner works. I'm fine with this as long as the other exercises are also converted when v3 goes live (as they otherwise wouldn't be able to run in the test runner).

Reply:
Ah! There needs to be backwards compatibility to existing v2 exercises?
I had been operating with the idea that the v2 exercises would become deprecated by v3 concepts.
If the former is the case, and assuming we _don't_ want to restructure all the existing exercises; then this structure needs to change a bit more - especially regarding the `.meta`directory which is not used in the C track in v2.

Perhaps @patricksjackson and @ryanplusplus  can give their opinion on this?

Reply:
> I had been operating with the idea that the v2 exercises would become deprecated by v3 concepts.

No not at all. The existing v2 exercise will all become Practice Exercises. See [this part of the FAQs](https://github.com/exercism/v3/blob/master/docs/maintainers/faqs.md#what-happens-to-the-existing-v2-exercises).

> Ah! There needs to be backwards compatibility to existing v2 exercises?

Well, they need to be aligned, as the test runner will have to be able to verify both Concept as Practice exercises. This does not necessarily mean that the old structure must be kept, it could also be that the existing v2 exercises are all bulk updated to a newer format. Whether that is something you'd like to take on, I'll that to you.